doc/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -68,14 +76,59 @@ The naming format and compression is governed by the script `/usr/local/bin/core
 
 Where `<comm>` value in the command name associated with a process. comm value of a running process can be read from `/proc/[pid]/comm` file
 
-## 4. Schema Additions
+## 4. Memory usage based techsupport invocation
+
+If the following condition resolves to true:
+```
+(mem_usage > mem_usage_threshold || ${container}_mem_usage > ${container}_mem_usage_threshold) || mem_free <= mem_free_threshold
+```
+
+where ```mem_usage```  is total system memory used (MemAvailable from /proc/meminfo),
+```mem_usage_threshold```  configured threshold, (100 - available_mem_threshold),
+
+```${container}_mem_usage```  used memory by $container ("docker stats --no-stream --format {{.MemUsage}}" $container),
+
+```${container}_mem_usage_threshold```  configured memory threshold for $container, (100 - ${container}available_mem_threshold), 
+
+```mem_free``` is the total minus mem usage, ```mem_free_threshold``` - mem free threshold.
+
+the SONiC techsupport is automatically generated.
+
+```mem_free_threshold``` is there to invoke dump when there is quite small amount of memory left that is needed to successfully execute "show techsupport". This is going to be 200 Mb by default, as at least 80-90 Mb takes "show techsupport" execution.

```

Comment:
>mem_free_threshold [](http://example.com/codeflow?start=3&length=18)

If suddenly mem_free_threshold < 80 MB, `show techsupport` will always fail. This invocation will make situation worse.

Reply:
@qiluo-msft If user configures too low mem_free_thrshold the "show techsupport" will fail. Currently there is no upper limit of "show techsupport" peak memory consumption, which will make it hard to verify the user does not configure this threshold too low.

> This invocation will make situation worse.

It is assumed that the situation is already critical and we try to take the dump as best effort.

BTW, the swap partition would make it at least possible to execute some commands, however, it is not configured in SONiC.

Reply:
Sorry for misleading. I am asking if the free memory suddenly drop to below 80 MB, show techsupport will always fail. This invocation will make situation worse.

Reply:
@qiluo-msft Do you mean that if suddenly an application perfroms a very big allocation so that we won't be able to perform "show techsupport". A similar situation might happen when the process leaks memory very quickly so that we don't have enough memory to inoke "show techsupport". In this design the dump collection is best effort, so we don't guaranty that the dump will be actually generated because of that. It is also assumed that if the system is running with that low free memory the situation is already critical (very likelly that reboot will happen) so that additional "show tech" does not help or make it worth, it is making the reboot happen earlier. If the user does want the system to stay with a very low free memory than the user has to disable this feature. Otherwise I don't see a way to handle such a situation.